### ISSUE TITLE
How to Windows screenshot when Android denys screenshots but Window screen shows content?

### ISSUE THREAD

**Issue by pstein** [Genymobile/scrcpy#5719]
It occurs sometimes that Android disallows taking a screenshot in certain Apps.

When I connect from Windows to Android and open such an App the App content is successfully shown in Windows scrcpy window.
Unfortunately scrcpy follows the no-screenshot rule and does not allow screenshots too.
Resp the resulting image is just black.

However the content must be available somewhere in Windows system (possibly in RAM storage).
Is there really no way to take a screenshot (outside of scrcpy) of such protected screen content?

**Comment by rom1v** [comment]
> It occurs sometimes that Android disallows taking a screenshot in certain Apps.

Yes, either using DRM (like Netflix) or enabling the "secure flag" (like lock screens or banking apps).

View with the "secure flag" enabled can no longer be captured by _shell_ since Android 12 (#36 #2129).

> When I connect from Windows to Android and open such an App the App content is successfully shown in Windows scrcpy window.

If it appears in the scrcpy window, then it is correctly captured, so you get the content. Btw, if you record (`scrcpy --record=file.mp4`), the content will be recorded.

> Unfortunately scrcpy follows the no-screenshot rule and does not allow screenshots too.

I have no idea what you are talking about. Either the image cannot be captured on Android, and it is black (including in the scrcpy window), either it can be captured, and you can record/screenshot it.
